pkg/storage/chunk/cache/groupcache.go

```diff
@@ -243,7 +243,7 @@ func (c *group) Store(ctx context.Context, keys []string, values [][]byte) error
 
 	var lastErr error
 	for i, key := range keys {
-		if err := c.cache.Set(ctx, key, values[i], time.Time{}, false); err != nil {
+		if err := c.cache.Set(ctx, key, values[i], time.Time{}, c.cacheType != stats.ChunkCache); err != nil {
```

Comment:
I think it'd be safer to use the `GetCacheType()` getter here since we wrap cache implementations

Reply:
updated.